api/dataservices/interface.go

```diff
@@ -0,0 +1,290 @@
+package dataservices
+
+// 	"github.com/portainer/portainer/api/dataservices"
+
+import (
+	"io"
+	"time"
+
+	"github.com/portainer/portainer/api/dataservices/errors"
+
+	portainer "github.com/portainer/portainer/api"
+)
+
+type (
+	// DataStore defines the interface to manage the data
+	DataStore interface {
+		Open() (newStore bool, err error)
+		Init() error
+		Close() error
+		MigrateData() error
+		Rollback(force bool) error
+		CheckCurrentEdition() error
+		BackupTo(w io.Writer) error
+		Export(filename string) (err error)
+		IsErrObjectNotFound(err error) bool
+
+		CustomTemplate() CustomTemplateService
+		EdgeGroup() EdgeGroupService
+		EdgeJob() EdgeJobService
+		EdgeStack() EdgeStackService
+		Endpoint() EndpointService
+		EndpointGroup() EndpointGroupService
+		EndpointRelation() EndpointRelationService
+		HelmUserRepository() HelmUserRepositoryService
+		Registry() RegistryService
+		ResourceControl() ResourceControlService
+		Role() RoleService
+		APIKeyRepository() APIKeyRepository
+		Settings() SettingsService
+		SSLSettings() SSLSettingsService
+		Stack() StackService
+		Tag() TagService
+		TeamMembership() TeamMembershipService
+		Team() TeamService
+		TunnelServer() TunnelServerService
+		User() UserService
+		Version() VersionService
+		Webhook() WebhookService
+	}
+
+	// CustomTemplateService represents a service to manage custom templates
+	CustomTemplateService interface {
+		GetNextIdentifier() int
+		CustomTemplates() ([]portainer.CustomTemplate, error)
+		CustomTemplate(ID portainer.CustomTemplateID) (*portainer.CustomTemplate, error)
+		Create(customTemplate *portainer.CustomTemplate) error
+		UpdateCustomTemplate(ID portainer.CustomTemplateID, customTemplate *portainer.CustomTemplate) error
+		DeleteCustomTemplate(ID portainer.CustomTemplateID) error
```

Comment:
why not go with the create rename and call it, List, Get, Update, Delete ?

Reply:
mostly because rebasing this was a constant regret - I almost removed it

when we have a faster merge times, i'll happily rename our methods to stop the repetition we have.